src/core/storage.rs

```diff
@@ -42,6 +42,8 @@ pub enum RecordSource {
 	Onchain,
 	/// For offchain data.
 	Offchain,
+	/// For persistent data (not pruned automatically)
+	Persistent,
```

Comment:
What do we want to use this for ? Also, I don't think this is a good place for it. This represents the source of the record, not its lifetime.

Reply:
We want this for session information, that should not be prunned. However, placement it in the RecordSource is really a logical mistake. So we can either tell that it is not a `RecordSource` but `RecordType` or add an additional enum to distinguish block based and persistent data.